daemon/event.go

```diff
@@ -466,6 +477,86 @@ func (d *daemon) closeEvents() error {
 	return nil
 }
 
+// StressEvent is making requests to daemon to see how daemon can handle them
+func (d *daemon) StressEvent(ctx context.Context, req *pb.TestEventLoadReq) (*pb.TestEventLoadResp, error) {
+	user, err := getUserFromIncomingContext(ctx)
+	if err != nil {
+		return &pb.TestEventLoadResp{}, err
+	}
+	if !user.SuperUser {
+		return &pb.TestEventLoadResp{}, NoPrivilegeToStressTest
+	}
+
+	ev, err := d.eventPool.GetEvent(store.Tag(req.EventName))
+	if ev == nil {
+		return &pb.TestEventLoadResp{}, fmt.Errorf("no such an event called %s, error: %v !", req.EventName, err)
+	}
+
+	if ev.GetConfig().Capacity < int(req.NumberOfTeams) {
+		return &pb.TestEventLoadResp{}, errors.New("event capacity is less than provided number of teams. skipping testing...")
+	}
+	var port, protocol string
+	if d.conf.Certs.Enabled {
+		port = strconv.FormatUint(uint64(d.conf.Port.Secure), 10)
+		protocol = "https://"
+	} else {
+		port = strconv.FormatUint(uint64(d.conf.Port.InSecure), 10)
+		protocol = "http://"
+	}
+	endPoint := fmt.Sprintf(protocol + req.EventName + "." + d.conf.Host.Http + ":" + port + "/signup")
+	resp := make(chan string)
+	for i := 0; i < int(req.NumberOfTeams); i++ {
+		go func() {
+			resp <- d.postRequest(endPoint, protocol)
+		}()
+	}
+	response := <-resp
+
+	return &pb.TestEventLoadResp{SignUpResult: response}, nil
```

Comment:
I am not sure this is the correct implementation. 
Since `resp` is a channel, we can have multiple input on it (the loop confirm that), so i think we need to have to send multiple response to the client. in this case we need a `stream` in the proto file and change the code a little bit. does it make sense?

Reply:
Yesss, I do agree, it will be stream based ;) I am aware of it 😉

And more, I think, having time based requests could be nice as well, like having list of times 

- [1, 5, 10, 15, 20] 

and see when the problem occurs mostly in case of an event. 


Currently the problem is that, when number of users who would like to signup bigger than queue, haaukins does not wait until a new lab is created, throws no labs available on queue error. It has cons and pros that we can discuss @gianmarcomennecozzi

Reply:
I got it. we can actually tests it with the number of labs present in the buffer queue andt then create a test event with an `availability` of 20 perhaps.